src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -116,9 +116,8 @@ export const SpawnHostModal: React.VFC<SpawnHostModalProps> = ({
   const awsRegions = awsData?.awsRegions;
   const volumes = volumesData?.myVolumes ?? [];
 
-  // When the modal is opened, reset to the "default" state of the Modal contents.
+  // When the modal is opened, pre-fill parts of the form (AWS region, public key, and expiration).
   useEffect(() => {
```

Comment:
The `dispatch({ type: "reset" });` was causing a race condition with the [`useEffect`](https://github.com/evergreen-ci/spruce/blob/main/src/pages/spawn/spawnHost/spawnHostModal/SetupScriptForm.tsx#L53-L62) here when spawning a host from a task. We would initially correctly pre-fill the distro based on the URL params, but then we would end up resetting it to be empty.

Besides, I think it makes more sense to reset the form when the modal is closed, not when it is opened.